src/ui/src/js/controllers/request_view.js

```diff
@@ -33,6 +33,33 @@ export default function requestViewController(
     request) {
   $scope.service = RequestService;
 
+ $scope.download = function() {
+      var filename = $scope.data.id;
+      var output;
+      var outputType = $scope.data.output_type;
+      if($scope.data.output === undefined || $scope.data.output == null){
+        window.alert("No output to download");
+        return;
+      } else if(outputType.toLowerCase() == "string"){
+        output = $scope.rawOutput;
+        filename = filename+".txt"
+      } else if(outputType.toLowerCase() == "html"){
+        output = $scope.htmlOutput;
+        filename = filename+".html"
+      } else if(outputType.toLowerCase() == "json"){
+        output = $scope.jsonOutput;
+        filename = filename+".json"
+      }
```

Comment:
It's definitely good to use the output type to determine the file extension. But I think you can simplify this and just use `$scope.rawOutput` everywhere. The other things are for presentation / formatting, and if someone is downloading the output we need to give them the *real* (raw) output, not the version we modified to look pretty for them.